src/System.Drawing.Common/src/System/Drawing/GdiplusNative.Unix.cs

```diff
@@ -21,24 +22,48 @@ internal unsafe partial class Gdip
 
             private static IntPtr LoadNativeLibrary()
             {
+                string libraryName = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "libgdiplus.dylib" : "libgdiplus.so";
+
                 IntPtr lib = IntPtr.Zero;
                 if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                 {
-                    lib = Interop.Libdl.dlopen("libgdiplus.dylib", Interop.Libdl.RTLD_NOW);
+                    lib = Interop.Libdl.dlopen(libraryName, Interop.Libdl.RTLD_NOW);
                 }
                 else
                 {
                     // Various Unix package managers have chosen different names for the "libgdiplus" shared library.
                     // The mono project, where libgdiplus originated, allowed both of the names below to be used, via
                     // a global configuration setting. We prefer the "unversioned" shared object name, and fallback to
                     // the name suffixed with ".0".
-                    lib = Interop.Libdl.dlopen("libgdiplus.so", Interop.Libdl.RTLD_NOW);
+                    lib = Interop.Libdl.dlopen(libraryName, Interop.Libdl.RTLD_NOW);
                     if (lib == IntPtr.Zero)
                     {
                         lib = Interop.Libdl.dlopen("libgdiplus.so.0", Interop.Libdl.RTLD_NOW);
                     }
                 }
 
+                // If we couldn't find libgdiplus in the system search path, try to look for libgdiplus in the
+                // NuGet package folders. This matches the DllImport behavior.
+                if (lib == IntPtr.Zero)
+                {
+                    string[] searchDirectories = ((string)AppContext.GetData("NATIVE_DLL_SEARCH_DIRECTORIES")).Split(':');
+
+                    foreach (var searchDirectory in searchDirectories)
+                    {
+                        var searchPath = Path.Combine(searchDirectory, libraryName);
+
+                        if (File.Exists(searchPath))
```

Comment:
Do you really need to call `File.Exists` here? It should work fine without it.

Reply:
You're right, I removed the `File.Exists`.